elasticsearch/templates/statefulset.yaml

```diff
@@ -216,3 +216,50 @@ spec:
         {{- if .Values.extraVolumeMounts }}
 {{ tpl .Values.extraVolumeMounts . | indent 10 }}
         {{- end }}
+      {{- if eq .Values.roles.master "true" }}
+      - name: elasticsearch-master-graceful-termination-handler
+        image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+        imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+        command:
+        - "sh"
+        - -c
+        - |
+          #!/usr/bin/env bash
+          set -eo pipefail
+
+          http () {
+              local path="${1}"
+              if [ -n "${ELASTIC_USERNAME}" ] && [ -n "${ELASTIC_PASSWORD}" ]; then
+                BASIC_AUTH="-u ${ELASTIC_USERNAME}:${ELASTIC_PASSWORD}"
+              else
+                BASIC_AUTH=''
+              fi
+              curl -XGET -s -k --fail ${BASIC_AUTH} {{ .Values.protocol }}://{{ template "masterService" . }}:{{ .Values.httpPort }}${path}
+          }
+
+          cleanup () {
+            while true ; do
+              local master="$(http "/_cat/master")"
+              if [[ $master == *"{{ template "uname" . }}"* && $master != *"${NODE_NAME}"* ]]; then
```

Comment:
With the suggestion above we can check that the `${NODE_NAME}` directly matches the `$master` without any wildcards. This potentially prevents any false positives if someone were to have more than 10 masters. E.g. `master-1*` matching `master-1` and `master-10`. 

This could then become something like (not tested):

```
local master="$(http "/_cat/master?h=node")"
if [[ "$master" == "{{ template "uname" . }}"* && "$master" != "${NODE_NAME}" ]]; then
```